torchao/quantization/quantize_/workflows/float8/float8_tensor.py

```diff
@@ -361,6 +427,37 @@ def _(func, types, args, kwargs):
         )
 
 
+def _float8_matmul_impl(
```

Comment:
how about

1. define _float8_mm_impl as the float8 version of `torch.mm`, as the lowest level shared code of maybe quantizing the input and then choosing a gemm
2. all other functions (matmul, linear, etc) call _float8_mm_impl

it's a bit confusing to have two different paths for linear and matmul